emily/handler/src/common/error.rs

```diff
@@ -128,21 +97,130 @@ pub enum Error {
+
+    /// This happens if the withdrawal entry that was stored in the database
+    /// was invalid, or if the withdrawal entry that we are creating to store
+    /// in the database is invalid.
+    #[error("withdrawal entry failed validation; {0}; ID: {1}")]
+    WithdrawalEntry(&'static str, WithdrawalEntryKey),
+
+    /// This happens when there is a mismatch in the request ID of the new
+    /// withdrawal event and the fetched withdrawal entry. Seeing this is
+    /// probably due to a programming error.
+    #[error("mismatch when updateing withdrawal request; existing: {0}; update: {1}")]
+    WithdrawalUpdate(WithdrawalEntryKey, u64),
+
+    /// This means that the stacks address in the environment for the
+    /// signers multisig address is invalid.
+    #[error("could not parse a stacks address from a string")]
+    InvalidStacksAddress(#[source] clarity::vm::errors::Error),
+
+    /// This happens when the request to DynamoDB succeeds but does not
+    /// return any values. This happens when the request instructs the
+    /// database to refrain from returning values, so this is likely a
+    /// programming error.
+    #[error("entry in database for deposit request not returned from DynamoDB; {0}")]
+    MissingAttributesDeposit(DepositEntryKey),
+
+    /// This happens when the request to DynamoDB succeeds but does not
+    /// return any values. This happens when the request instructs the
+    /// database to refrain from returning values, so this is likely a
+    /// programming error.
+    #[error("entry in database for withdrawal request not returned from DynamoDB; {0}")]
+    MissingAttributesWithdrawal(WithdrawalEntryKey),
+
+    /// DynamoDB should only contain one entry per withdrawal request ID.
+    ///
+    /// TODO: In case of a re-org, tripple check that we can identify the
+    /// correct withdrawal request if the transaction is replayed.
+    #[error("DynamoDB contained many entries for the given request ID: {0}")]
+    TooManyWithdrawalEntries(u64),
+
+    /// This happens when we fail to decode a base64 encoded string into a
+    /// vector of bytes.
+    #[error("failed to base64 decode the string into bytes; {0}")]
+    Base64Decode(base64::DecodeError),
+
+    /// This is used when trying to get a required value from the
+    /// environment and that operation fails.
+    #[error("could not read the environment variable; {0}")]
+    EnvVariable(#[from] env::VarError),
+
+    /// This occurs when serializing or deserializing an object into or
+    /// from JSON.
+    #[error("{0}")]
+    SerdeJson(#[from] serde_json::Error),
+
+    /// This occurs when converting structs to and from DynamoDB objects.
+    #[error("{0}")]
+    SerdeDynamo(#[from] serde_dynamo::Error),
+
+    /// This happens when attempting to parse an integer from a string that
+    /// has been read from an environment variable.
+    #[error("could not parse the string into an integer; {0}")]
```

Comment:
Add some note that this string come from env var ?

Reply:
Updated in https://github.com/stacks-sbtc/sbtc/pull/1627/commits/e20b276df8646d7a1d3739e5465913fb6ad3169e